soc/nordic/validate_binding_headers.c

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/sys/util_macro.h>
+#include <zephyr/toolchain.h>
+
+#include <nrf.h>
+
+/**
+ * Check that a definition from a dt-bindings header matches the correct value
+ * from MDK, which may be provided in the form of a C type.
+ *
+ * Note: all dt-bindings headers which have been included by DTS in this build
+ * are automagically included in this file as well. See CMakeLists.txt.
+ *
+ * @param dt_val devicetree-specific value.
+ * @param mdk_val MDK-provided value.
+ */
+#define CHECK_EQUAL_IFDEF(dt_val, mdk_val)                                                         \
+	IF_ENABLED(dt_val,                                                                         \
```

Comment:
This will cause the assert to appear only if `dt_val` is defined and equal to 1?
`!IS_EMPTY` could work for here better?

Reply:
Oops. Yeah, `IF_ENABLED` didn't work, and neither will `IS_EMPTY`, because it returns 1 for undefined arguments.

I settled on using `#if defined`, which doesn't look great, but it works as intended.

I also considered fixing this by adding `IS_EQ(dt_val, dt_val)`, which would check if `dt_val` is defined as an integer between 0--4095. At least this would cover the size of a typical register block, but some values would have to be changed from hex to decimal.